hypershift-operator/ignitionserver/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+// This is the simplest http server that enable us to satisfy
+// 1 - 1 relation between clusters and ign endpoints.
+// At the moment this acts as purely a proxy towards MCS that are created on demand by NodePools,
+// this might / might not change in the future.
+// TODO (alberto): Support only https.
+// TODO (alberto): Support only token authenticated requests.
+// TODO (alberto): Parameterize listening port and URL to proxy.
+// TODO (alberto): Metrics.
+
+type ignHandler struct{}
+
+var _ http.Handler = &ignHandler{}
+
+func (h *ignHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// We only match /ign/$NodePoolName
+	ignPathPattern, err := regexp.Compile("^/ign/[^/ ]*$")
```

Comment:
Can extract this into a package constant with MustCompile to avoid error handling here